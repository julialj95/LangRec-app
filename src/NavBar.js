import React, { Component } from "react";
import TokenService from "./services/token-service";
import { Link } from "react-router-dom";
import { LangrecContext } from "./LangrecContext";

class NavBar extends Component {
  static contextType = LangrecContext;
  handleLogoutClick = () => {
    TokenService.clearAuthToken();
    this.context.handleLoginChange();
  };

  renderLoggedInDisplay() {
    return (
      <>
        <Link to="/saved-resources">My Saved Resources</Link>
        <Link to="/share">Submit Resources</Link>
        <Link to="/" onClick={this.handleLogoutClick}>
          Logout
        </Link>
      </>
    );
  }

  renderLoggedOutDisplay() {
    return (
      <>
        <Link to="/signup">Sign Up</Link>
        <Link to="/login">Log In</Link>
      </>
    );
  }

  render() {
    return (
      <nav>
        <ul className="navbar">
          <Link to="/">Home</Link>
          <Link to="/recs">Get Recommendations</Link>
          {TokenService.hasAuthToken()
            ? this.renderLoggedInDisplay()
            : this.renderLoggedOutDisplay()}
        </ul>
      </nav>
    );
  }
}

export default NavBar;
