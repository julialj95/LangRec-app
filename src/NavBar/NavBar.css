@import url("https://fonts.googleapis.com/css2?family=Nunito&display=swap");

.navbar {
  display: flex;
  flex-direction: column;
  align-content: space-around;
  text-align: center;
  padding: 5px;
}

a {
  text-decoration: none;
  font-family: "Nunito", sans-serif;
  flex-grow: 1;
  color: white;
}

@media all and (min-width: 600px) {
}
@media all and (min-width: 768px) {
  .navbar {
    flex-direction: row;
    justify-content: space-around;
    flex-wrap: wrap;
  }
}
@media all and (min-width: 992px) {
  .navbar {
    flex-wrap: nowrap;
  }
}

@media all and (min-width: 1200px) {
}
