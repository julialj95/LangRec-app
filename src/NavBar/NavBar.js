import React, { Component } from "react";
import TokenService from "../services/token-service";
import { Link } from "react-router-dom";
import { LangrecContext } from "../LangrecContext";
import "./NavBar.css";

class NavBar extends Component {
  static contextType = LangrecContext;
  handleLogoutClick = () => {
    TokenService.clearAuthToken();
    this.context.handleLoginChange(false);
    this.context.handleRecommendedResources([]);
    this.context.handleSavedResourcesOnLogout();
  };

  renderLoggedInDisplay() {
    return (
      <>
        <Link className="navlink" to="/saved-resources">
          MY SAVED RESOURCES
        </Link>
        <Link className="navlink" to="/" onClick={this.handleLogoutClick}>
          LOGOUT
        </Link>
      </>
    );
  }

  renderLoggedOutDisplay() {
    return (
      <>
        <Link className="navlink" to="/signup">
          SIGN UP
        </Link>
        <Link className="navlink" to="/login">
          LOG IN
        </Link>
        <Link className="navlink" to="/demo">
          DEMO
        </Link>
      </>
    );
  }

  render() {
    return (
      <nav>
        <ul className="navbar">
          <Link className="navlink" to="/">
            HOME
          </Link>
          <Link className="navlink" to="/recs">
            GET RECOMMENDATIONS
          </Link>
          {this.context.isLoggedIn
            ? this.renderLoggedInDisplay()
            : this.renderLoggedOutDisplay()}
        </ul>
      </nav>
    );
  }
}

export default NavBar;
