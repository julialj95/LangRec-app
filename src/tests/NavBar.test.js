import React from "react";
import ReactDOM from "react-dom";
import NavBar from "../NavBar/NavBar";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(
    <Router>
      <NavBar />
    </Router>,
    div
  );

  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create(
      <Router>
        <NavBar />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
