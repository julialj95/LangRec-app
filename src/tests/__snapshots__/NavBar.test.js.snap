// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<nav>
  <ul
    className="navbar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      HOME
    </a>
    <a
      href="/recs"
      onClick={[Function]}
    >
      GET RECOMMENDATIONS
    </a>
    <a
      href="/signup"
      onClick={[Function]}
    >
      SIGN UP
    </a>
    <a
      href="/login"
      onClick={[Function]}
    >
      LOG IN
    </a>
  </ul>
</nav>
`;
